src/ejabberd_sup.erl

```diff
@@ -194,10 +194,18 @@ cets_specs() ->
 
 cets_specs(disabled) ->
     [];
-cets_specs(#{backend := DiscoBackend, cluster_name := ClusterName,
-             nodelist_file := NodeFile}) ->
-    ConfigDir = filename:dirname(mongoose_config:get_config_path()),
-    DiscoFile = filename:join(ConfigDir, NodeFile),
+cets_specs(#{backend := DiscoBackend, cluster_name := ClusterName} = Opts) ->
+    DiscoFile =
+        case {DiscoBackend, Opts} of
+            {file, #{nodelist_file := NodeFile}} ->
+                NodeFile;
+            {file, _} ->
+                ?LOG_CRITICAL(#{what => nodelist_file_option_is_required,
```

Comment:
Just a comment: we could check it with a `process` function (for the entire CETS section) in the config spec, and leverage the error handling of the config parser.